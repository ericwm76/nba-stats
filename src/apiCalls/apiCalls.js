const baseUrl = 'https://www.balldontlie.io/api/v1'

export const searchByName = async (firstName, lastName) => {
  let response = await fetch(`${baseUrl}/players/?search=${firstName}%20${lastName}&per_page=100`);

  if(!response.ok) {

  }

  let data = await response.json();

  return data.data
}

export const getPlayerSeasonAvgs = async (id) => {
  let response = await fetch(`${baseUrl}/season_averages?player_ids[]=${id}`);

  if(!response.ok) {

  }

  let data = await response.json();

  return data.data;
}

export const getGameByGameStats = async (id) => {
  let response = await fetch(`${baseUrl}/stats?per_page=100&seasons[]=2019&player_ids[]=${id}`)

  if(!response.ok) {

  }

  let data = await response.json();
  console.log(data.data)

  return data.data;
}