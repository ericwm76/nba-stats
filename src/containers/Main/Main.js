import './Main.scss';
import React from 'react';
import { Link } from 'react-router-dom';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import {  } from '../../actions';
import PropTypes from 'prop-types';

export const Main = () => {


  return (
    <div className="Main">
      
    </div>
  )
}

const mapStateToProps = state => ({

})

const mapDispatchToProps = dispatch => (
  bindActionCreators({

  }, dispatch)
)

export default connect(mapStateToProps, mapDispatchToProps)(Main)