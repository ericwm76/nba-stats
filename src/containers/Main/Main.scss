@import url('https://fonts.googleapis.com/css?family=Rajdhani:400,700&display=swap');

.Main {
  background-color: #212121;
  color: white;
  font-family: "Rajdhani", sans-serif;  
  height: 90vh;
  padding: 20px;
  text-align: center;
  button {
    background-color: #212121;
    border: 1px solid white;
    border-radius: 3px;
    color: white;
    display: block;
    font-family: "Rajdhani", sans-serif;  
    font-size: .9em;
    height: 44px; 
    margin: 5px;
    width: 10%;
  }
}