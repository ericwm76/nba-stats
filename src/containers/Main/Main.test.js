import React from 'react';
import { shallow } from 'enzyme';
import { Main, mapStateToProps } from './Main';
import { getGameByGameStats } from '../../apiCalls/apiCalls';

jest.mock('../../apiCalls/apiCalls.js')

describe('Main', ( )=> {
  let wrapper, mockPlayers, mockEvent, mockStats;

  beforeEach(() => {
    mockEvent = {
      preventDefault: jest.fn()
    }

    mockPlayers = [
      {
        first_name: "Lebron",
        last_name: "James",
        height_feet: 6,
        height_inches: 8,
        weight_pounds: 250,
        position: "F",
        id: 237,
        team: {
          full_name: "Los Angeles Lakers",
        }
      }
    ]

    mockStats = [
      { 0: 
        {
        "id":29,
        "ast":2,
        "blk":2,
        "dreb":8,
        "fg3_pct":0.25,
        "fg3a":4,
        "fg3m":1,
        "fg_pct":0.429,
        "fga":21,
        "fgm":9,
        "ft_pct":0.8,
        "fta":5,
        "ftm":4,
        "game":{
          "id":1,
          "date":"2018-10-16T00:00:00.000Z",
          "home_team_id":2,
          "home_team_score":105,
          "season":2018,
          "visitor_team_id":23,
          "visitor_team_score":87
        },
        "min":"36:49",
        "oreb":2,
        "pf":3,
        "player":{
          "id":246,
          "first_name":"Lebron",
          "last_name":"James",
          "position":"F",
          "team_id":23
        },
        "pts":23,
        "reb":10,
        "stl":1,
        "team":{
          "id":23,
          "abbreviation":"LAL",
          "city":"Los Angeles",
          "conference":"West",
          "division":"Pacific",
          "full_name":"Los Angeles Lakers",
          "name":"Lakers"
        },
        "turnover":5
        }
      }
    ]

    getGameByGameStats.mockImplementation((id) => {
      return Promise.resolve(mockStats)
    })

    wrapper = shallow(<Main players={mockPlayers}/>)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should call comparePlayers on click', () => {
    wrapper.instance().comparePlayers = jest.fn()

    wrapper.find('button').simulate('click', mockEvent);

    expect(wrapper.instance().comparePlayers).toHaveBeenCalled();
  })

  it('should call getGameByGameStats when comparePlayers is called', () => {
    wrapper.instance().comparePlayers(mockEvent, mockPlayers);

    expect(getGameByGameStats).toHaveBeenCalled();
  })

  it('should update state when comparePlayers is called', async () => {
    const expected = [ { 237: mockStats } ]
    await wrapper.instance().comparePlayers(mockEvent, mockPlayers);
    expect(wrapper.state('gameStats')).toEqual(expected)
  })
})