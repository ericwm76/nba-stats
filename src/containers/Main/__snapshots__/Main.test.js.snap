// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main should match the snapshot 1`] = `
<div
  className="Main"
>
  <button
    onClick={[Function]}
  >
    Compare Players
  </button>
  <Card
    key="237"
    player={
      Object {
        "first_name": "Lebron",
        "height_feet": 6,
        "height_inches": 8,
        "id": 237,
        "last_name": "James",
        "position": "F",
        "team": Object {
          "full_name": "Los Angeles Lakers",
        },
        "weight_pounds": 250,
      }
    }
  />
  0
</div>
`;
